File: src/go/readdir_test.go
package main

import (
	"fmt"
	"os"
)

func main() {
	fmt.Println("--- ReadDir Test ---")
	entries, err := os.ReadDir(".")
	if err != nil {
		fmt.Println("Error reading dir:", err)
		return
	}

	for _, e := range entries {
		fmt.Println("Entry:", e.Name())
	}
}

File: src/python/readdir_test.py
import os

def main():
    print("--- Readdir Test ---")
    
    try:
        entries = os.listdir("/")
        print("Directory listing for /:")
        for entry in sorted(entries):
            print(f"  {entry}")
    except Exception as e:
        print(f"Failed to read directory: {e}")

if __name__ == "__main__":
    main()
